jerry-core/api/jerry.c

```diff
@@ -6428,6 +6429,140 @@ jerry_get_container_type (const jerry_value_t value) /**< the container object *
   return JERRY_CONTAINER_TYPE_INVALID;
 } /* jerry_get_container_type */
 
+/**
+ * Return the underlying buffer from a Container or a Container Iterator.
+ * Sets the boolean input value to `true` if the container object is a
+ * key-value structure and `false` if not.
+ *
+ * Note:
+ *     the returned value must be freed with a jerry_release_value call
+ *
+ * @return jerry_value_t representing an array containing the elemets of the map.
+ *         Empty array if the Iterator is finished/empty.
+ */
+jerry_value_t
+jerry_get_buffer_from_container (jerry_value_t value, /**< the container or iterator object */
+                                bool *is_key_value_p) /**< [out] is key-value structure */
+{
+  jerry_assert_api_available ();
+
+#if JERRY_BUILTIN_CONTAINER
+  ecma_collection_t *collection_buffer = ecma_new_collection ();
+
+  if (!ecma_is_value_object (value))
+  {
+    return false;
+  }
+
+  ecma_object_t *obj_p = ecma_get_object_from_value (value);
+  ecma_extended_object_t *ext_obj_p = (ecma_extended_object_t *) obj_p;
+  ecma_object_type_t type = ecma_get_object_type (obj_p);
+
+  if (jerry_get_container_type (value) == JERRY_CONTAINER_TYPE_INVALID &&
+        !(type == ECMA_OBJECT_TYPE_CLASS &&
+          (ext_obj_p->u.cls.type == ECMA_OBJECT_CLASS_MAP_ITERATOR ||
+           ext_obj_p->u.cls.type == ECMA_OBJECT_CLASS_SET_ITERATOR)))
+  {
+    ecma_collection_free (collection_buffer);
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG ("Value is not a Container or Iterator")));
+  }
+
+  ecma_collection_t *container_p = ECMA_GET_INTERNAL_VALUE_POINTER (ecma_collection_t, ext_obj_p->u.cls.u3.value);
+  lit_magic_string_id_t lit_id = ext_obj_p->u.cls.u2.container_id;
+  uint32_t entry_count =  ECMA_CONTAINER_ENTRY_COUNT (container_p);
+  uint8_t entry_size = ecma_op_container_entry_size (lit_id);
+
+  uint32_t index = 0;
+  uint8_t iterator_kind = ECMA_ITERATOR__COUNT;
+  ecma_value_t *start_p = ECMA_CONTAINER_START (container_p);
+  bool is_value_iterator = false;
+
+  *is_key_value_p = false;
+
+  if (type == ECMA_OBJECT_TYPE_CLASS &&
+      (ext_obj_p->u.cls.type == ECMA_OBJECT_CLASS_MAP_ITERATOR ||
+        ext_obj_p->u.cls.type == ECMA_OBJECT_CLASS_SET_ITERATOR))
+  {
+    is_value_iterator = true;
+
+    ecma_value_t iterated_value = ext_obj_p->u.cls.u3.iterated_value;
+
+    if (ecma_is_value_empty (iterated_value))
+    {
+      return ecma_op_new_array_object_from_collection (collection_buffer, false);
+    }
+
+    ecma_extended_object_t *map_object_p = (ecma_extended_object_t *) (ecma_get_object_from_value (iterated_value));
+    lit_id = map_object_p->u.cls.u2.container_id;
```

Comment:
AFAIK, this can be removed and the `map_object_p->u.cls.u2.container_id` can be used directly in line 6507.